src/Hooks/WorkerEventListener.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Laravel\Nightwatch\Hooks;
+
+use Illuminate\Queue\Events\JobPopping;
+use Illuminate\Queue\Events\JobProcessing;
+use Illuminate\Queue\Events\Looping;
+use Illuminate\Queue\Events\WorkerStopping;
+use Laravel\Nightwatch\Core;
+use Laravel\Nightwatch\State\CommandState;
+use Throwable;
+
+/**
+ * @internal
+ */
+final class WorkerEventListener
+{
+    /**
+     * @param  Core<CommandState>  $nightwatch
+     */
+    public function __construct(
+        private Core $nightwatch,
+    ) {
+        //
+    }
+
+    public function __invoke(Looping|JobPopping|JobProcessing|WorkerStopping $event): void
+    {
+        try {
+            match ($event::class) {
+                Looping::class, WorkerStopping::class => $this->nightwatch->ingest(),
+                JobPopping::class => $this->nightwatch->resetStateForNextJob(),
+                JobProcessing::class => $this->nightwatch->prepareForJob($event->job),
+            };
```

Comment:
To chat about: what if we completely removed the need for `resetStateForNextJob`.